File: app/src/main/java/com/example/newsapp/data/api/NetworkService.java
package com.example.newsapp.data.api;

import com.example.newsapp.data.datamodels.DataResponse;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface NetworkService {

    @GET("everything")
    Call<DataResponse> getAllNews(@Query("q") String query,
                                  @Query("apiKey") String apiKey);

    @GET("everything")
    Call<DataResponse> getNewsByDomains(@Query("domains") String domains,
                                        @Query("apiKey") String apiKey);

    @GET("sources")
    Call<DataResponse> getSources(@Query("apiKey") String apiKey);

    @GET("top-headlines")
    Call<DataResponse> getTopHeadlines(@Query("country") String country,
                                       @Query("apiKey") String apiKey);

    @GET("top-headlines")
    Call<DataResponse> getTopHeadlinesByCountryCategory(@Query("country") String country,
                                                @Query("category") String category,
                                                @Query("apiKey") String apiKey);

    @GET("top-headlines")
    Call<DataResponse> getTopHeadlineBySearch(@Query("q") String query,
                                              @Query("apiKey") String apiKey);


}

File: app/src/main/java/com/example/newsapp/data/api/NetworkService.kt
package com.example.newsapp.data.api

import com.example.newsapp.data.datamodels.DataResponse
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface NetworkService {

    @GET("everything")
    fun getAllNews(@Query("q") query: String,
                   @Query("apiKey") apiKey: String): Call<DataResponse>

    @GET("everything")
    fun getNewsByDomains(@Query("domains") domains: String,
                         @Query("apiKey") apiKey: String): Call<DataResponse>

    @GET("sources")
    fun getSources(@Query("apiKey") apiKey: String): Call<DataResponse>

    @GET("top-headlines")
    fun getTopHeadlines(@Query("country") country: String,
                        @Query("apiKey") apiKey: String): Call<DataResponse>

    @GET("top-headlines")
    fun getTopHeadlinesByCountryCategory(@Query("country") country: String,
                                         @Query("category") category: String,
                                         @Query("apiKey") apiKey: String): Call<DataResponse>

    @GET("top-headlines")
    fun getTopHeadlineBySearch(@Query("q") query: String,
                               @Query("apiKey") apiKey: String): Call<DataResponse>


}
